### ISSUE TITLE
[Bug] TypeError: (0 , common_1.errorMessage) is not a function

### ISSUE THREAD

**Issue by colinking** [temporalio/sdk-typescript#900]
### What are you really trying to do?

Installing version `1.3.0` of `temporalio/worker` should install `1.3.0` of `temporalio/common` rather than the latest minor version. Ideally each Temporal SDK pins their Temporal dependencies.

### Describe the bug

Attempting to install an older version of the Temporal SDK will pull in the latest minor version of the underlying packages, e.g. `@temporalio/common`. This causes issues since there have been breaking changes in minor releases. For example, `1.4.0` unexported `errorMessage` so version `1.3.0` of `temporalio/worker` errors when trying to import it:

```
error: TypeError: (0 , common_1.errorMessage) is not a function
       at /airplane/.airplane/node_modules/@temporalio/worker/lib/tracing.js:70:102
       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
       at async /airplane/.airplane/node_modules/@temporalio/worker/lib/worker.js:554:24
 }
```

### Minimal Reproduction

Create a `package.json` that depends on `1.3.0` like so:

```
{
  "dependencies": {
    "@temporalio/worker": "1.3.0",
    "@temporalio/workflow": "1.3.0"
  }
}
```

Run `npm install`. Inspect `package-lock.json` and notice it installs a mixture of `1.3.0` and `1.4.0` versions of the various Temporal SDK packages:

```
▶ cat package-lock.json | grep -A5 temporal
    "@temporalio/workflow": {
      "version": "1.3.0",
      "resolved": "https://registry.npmjs.org/@temporalio/workflow/-/workflow-1.3.0.tgz",
      "integrity": "sha512-iK6LkQLF5vN9R+eONZK+LGR+x5ZgFTc+mKigqNzNZFwLsAm4WM8MmBvh5uKCsIXdfN2F4tnxB/I6svUN6OZBdA==",
      "requires": {
        "@temporalio/common": "^1.3.0",
        "@temporalio/internal-workflow-common": "^1.3.0",
        "@temporalio/proto": "^1.3.0"
      }
    },

...


    "@temporalio/common": {
      "version": "1.4.0",
      "resolved": "https://registry.npmjs.org/@temporalio/common/-/common-1.4.0.tgz",
      "integrity": "sha512-94TGVWh4iUQt86iojN/tc02Au4m+3WKXK4YVjYKxD33KTL0J7VYW1cZq3ELdicUNhVP35ioAr88K6Vs1KNPyLA==",
      "requires": {
        "@opentelemetry/api": "^1.0.3",
        "@temporalio/proto": "^1.4.0",
        "long": "^5.2.0",
        "ms": "^2.1.3",
        "proto3-json-serializer": "^1.0.3",
        "protobufjs": "^7.0.0"
      }
```

Since `@temporalio/workflow` depends on `"@temporalio/common": "^1.3.0"` instead of `"@temporalio/common": "1.3.0"`, it installs the latest minor version.

### Environment/Versions

- OS and processor: M1 Mac
- Temporal Version: 1.3.0
- Are you using Docker or Kubernetes or building Temporal from source? N/A

```
▶ npm version
  npm: '8.15.0',
  node: '18.7.0',
```

### Additional context

N/A

**Comment by bergundy** [comment]
Thanks for reporting this. We will address this in following releases.

**Comment by bergundy** [comment]
We will also publish a 1.3.1 patch

**Comment by colinking** [comment]
Fantastic. Thanks for the quick response, @bergundy.

**Comment by bergundy** [comment]
Fixed in 1.3.1 and 1.4.1

**Comment by colinking** [comment]
@bergundy this issue appears fixed in `1.3.1`, but not `1.4.1` nor `1.4.2`.

## 1.3.1

Uses `~` and only accepts patch version updates (correct):

<img width="960" alt="CleanShot 2022-10-11 at 20 21 39" src="https://user-images.githubusercontent.com/2907397/195221487-58935290-144f-4054-b111-871b4c435347.png">

This lockfile was generated from this `package.json`:

```json
{
  "dependencies": {
    "@temporalio/worker": "1.3.1",
    "@temporalio/workflow": "1.3.1"
  }
}
```


## 1.4.2

Uses `^` and will accept minor version updates (broken):

<img width="921" alt="CleanShot 2022-10-11 at 20 18 00" src="https://user-images.githubusercontent.com/2907397/195221275-4cf7529d-c729-448c-9b9c-a158baa9dcc6.png">

This lockfile was generated from this `package.json`:

```json
{
  "dependencies": {
    "@temporalio/worker": "1.4.2",
    "@temporalio/workflow": "1.4.2"
  }
}
```

**Comment by bergundy** [comment]
I know, we've updated the contributing guide explaining how to ping to minor.
Thanks for reporting.
